src/routes/NodeOverview/NodeSettings.tsx

```diff
@@ -0,0 +1,172 @@
+import { ConfigOptions } from '@ironfish/sdk'
+import {
+  Flex,
+  chakra,
+  TextField,
+  Button,
+  Skeleton,
+  useIronToast,
+  Box,
+  useColorModeValue,
+  NAMED_COLORS,
+} from '@ironfish/ui-kit'
+import useNodeSettings from 'Hooks/node/useNodeSettings'
+import { FC, useState, useEffect, memo } from 'react'
+import pick from 'lodash/pick'
+import DetailsPanel from 'Components/DetailsPanel'
+import AccountSettingsImage from 'Svgx/AccountSettingsImage'
+
+const Information: FC = memo(() => {
+  const textColor = useColorModeValue(
+    NAMED_COLORS.GREY,
+    NAMED_COLORS.LIGHT_GREY
+  )
+  return (
+    <Box maxWidth="21.5rem" mt="1rem">
+      <chakra.h3 mb="1rem">Settings</chakra.h3>
+      <chakra.h5 mb="2rem" color={textColor}>
+        Change your node settings to fit you needs
+      </chakra.h5>
+      <AccountSettingsImage />
+    </Box>
+  )
+})
+
+const SETTINGS_KEYS = [
+  'nodeName',
+  'blockGraffiti',
+  'nodeWorkers',
+  'minPeers',
+  'maxPeers',
+  'blocksPerMessage',
+]
+
+const NodeSettings: FC = () => {
+  const [nodeSettings, setNodeSettings] = useState<Partial<ConfigOptions>>({})
+  const {
+    data,
+    loaded,
+    actions: { saveSettings },
+  } = useNodeSettings()
+  const toast = useIronToast({
+    containerStyle: {
+      mb: '1rem',
+    },
+  })
+
+  useEffect(() => {
+    setNodeSettings(pick(data, SETTINGS_KEYS))
+  }, [data])
+
+  const updateSettingValue = (key: string, value: unknown) =>
+    setNodeSettings(prev => ({
+      ...prev,
+      [key]: value,
+    }))
+
+  const handleSaveSettings = () => {
+    saveSettings(nodeSettings).then(() => toast({ title: 'Settings saved' }))
+  }
+
+  const hasNoChanges = () => {
+    return !(
+      data &&
+      nodeSettings &&
+      SETTINGS_KEYS.some(
+        (key: keyof ConfigOptions) =>
+          data[key]?.toString() !== nodeSettings[key]?.toString()
+      )
+    )
+  }
```

Comment:
suggestion: I like to use `useMemo` for stuff like this since the value won't be recomputed unless the dependencies change.

E.g. 

```tsx
const hasNoChanges = useMemo(() => {
    return !(
      data &&
      nodeSettings &&
      SETTINGS_KEYS.some(
        (key: keyof ConfigOptions) =>
          data[key]?.toString() !== nodeSettings[key]?.toString()
      )
    )
}, [data, nodeSettings])
```